deploy/crds/upgrade.managed.openshift.io_upgradeconfigs_crd.yaml

```diff
@@ -157,22 +185,49 @@ spec:
                       - Upgraded
                       - Failed
                     type: string
+                  scalingFailed:
+                    description: A boolean advising if the upgrade scaling has failed.
+                    properties:
+                      dimension:
+                        enum:
+                          - up
+                          - down
+                        type: string
+                      failed:
+                        type: boolean
+                    type: object
                   startTime:
                     format: date-time
                     type: string
                   version:
                     description: Desired version of this upgrade
                     type: string
+                  windowBreached:
+                    description: A boolean advising if the UpgradeConfig window has been breached.
+                    type: boolean
                   workerCompleteTime:
                     format: date-time
                     type: string
                   workerStartTime:
                     format: date-time
                     type: string
+                  workerTimeout:
+                    description: A boolean advising if the clusters control plane upgrade has timed out.
+                    type: boolean
                 required:
                   - phase
                 type: object
               type: array
+            notificationEventSent:
```

Comment:
Not hugely familiar with this part of the code anymore but notifications on the status seems wrong to me

Reply:
I get what you mean. I'm kind of 50/50 on its incorporation into the Status. 

I tend to lean towards their inclusion here as being fine. On the one hand notifications are currently a core, product-agnostic aspect of the operator's behaviour. On the other hand, I regularly think they shouldn't be.
On the other-other hand, the notifications are an integral, standalone step of the OSD upgrader, which is why their representation here carries some weight.

Keen to engage in discussion about it though.

Reply:
I'm 100% against it being on the status. Recorded in a status condition would be an easy generic way of recording this.

Other options are to maybe have the notifications manage a configmap for state and remove from the upgrader altogether (again, been a while since I looked in there)

Reply:
Yeah, I should've seen your other note about status conditions before replying here. :) I agree on that.

Reply:
:+1: Sent this between notifications too :smile:

Reply:
Is there a way to make this field on visually present when .type= "OSD" ? This is kind of what i was thinking in regards to https://github.com/openshift/managed-upgrade-operator/pull/219/files#diff-82dc411a0ed86675e9efddaa43dadc4fc6a75ea0e672400429e8ea39424da44cR47-R68